sphinxcontrib/drawio.py

```diff
@@ -65,88 +64,123 @@ def boolean_spec(argument: Any) -> bool:
+        return image
+
+
+class DrawIOFigure(DrawIOBase, Figure):
+    option_spec = Figure.option_spec.copy()
+    option_spec.update(DrawIOBase.option_spec)
+
+    @staticmethod
+    def _get_image_node(nodes):
+        figure, = nodes                  # TODO: test
+        image, *caption = figure         # TODO: test
+        return image
+
+
+OPTIONAL_UNIQUES = {"export-height": "height",
+                    "export-width": "width"}
+
+
+def drawio_export(builder: Builder, options: dict, in_filename: str,
                   default_output_format: str) -> str:
     """Render drawio file into an output image file."""
 
-    page_index = str(node["config"].get("page-index", 0))
-    output_format = node["config"].get("format") or default_output_format
-    scale = str(node["config"].get("scale", self.config.drawio_default_scale))
-    transparent = node["config"].get("transparency", self.config.drawio_default_transparency)
-    no_sandbox = self.config.drawio_no_sandbox
+    page_index = str(options.get("page-index", 0))
+    output_format = options.get("format") or default_output_format
+    scale = str(options.get("export-scale",
+                            builder.config.drawio_default_export_scale) / 100)
+    transparent = options.get("transparency",
+                              builder.config.drawio_default_transparency)
+    no_sandbox = builder.config.drawio_no_sandbox
+
+    input_abspath = Path(in_filename)
+    input_relpath = input_abspath.relative_to(builder.srcdir)
+    input_stem = input_abspath.stem
 
     # Any directive options which would change the output file would go here
     unique_values = (
         # This ensures that the same file hash is generated no matter the build directory
         # Mainly useful for pytest, as it creates a new build directory every time
-        node["filename"].replace(self.builder.srcdir, ""),
+        str(input_relpath),
         page_index,
         scale,
-        output_format,
-        *[str(node["config"].get(option)) for option in DrawIO.optional_uniques]
+        'true' if transparent else 'false',
+        *[str(options.get(option)) for option in OPTIONAL_UNIQUES]
     )
     hash_key = "\n".join(unique_values)
     sha_key = sha1(hash_key.encode()).hexdigest()
-    filename = "drawio-{}.{}".format(sha_key, default_output_format)
-    file_path = posixpath.join(self.builder.imgpath, filename)
-    out_file_path = os.path.join(self.builder.outdir, self.builder.imagedir,
-                                 filename)
-
-    if (os.path.isfile(out_file_path)
-            and getmtime(in_filename) < getmtime(out_file_path)):
-        return file_path
-
-    ensuredir(os.path.dirname(out_file_path))
-
-    if self.builder.config.drawio_binary_path:
-        binary_path = self.builder.config.drawio_binary_path
+    export_dir = Path(builder.srcdir) / '.drawio' / sha_key
+    ensuredir(export_dir)
+    filename = Path(input_stem).with_suffix('.' + output_format)
```

Comment:
I'd prefer this not to just be `folder-1.png`. If possible:
* use the filename of the draw.io file; as the tests have it'd be `box`.
* if the options are changed from the config **defaults**, append them in alphabetical order.

e.g. for a file named box with the zoom changed from 100 to 150:
`box-scale150.png`

Reply:
I'm not sure I'm following you. `box.drawio` will be exported to `.drawio/<hash A>/box.png`. The HTML builder will copy that file to `_build/_html/_images/box.png`. If`box.drawio` is referenced in another `drawio-image` directive with different export options, it will be exported to ``.drawio/<hash B>/box.png`. The HTML builder copies that file to `_build/_html/_images/box1.png`.

I feel that adding the export options to the filename:
- is not so useful since in most cases you will only reference a diagram once,
- makes image filenames unnecessarily long, and
- outside the scope of this PR

Reply:
Okay then.
This is behaviour seems to be that of the image figure backend, yeah?

Reply:
Which behavior you are referring to? If you're talking about the copying of the image files, that is how the HTML builder handles image nodes.

Reply:
Yeah, that's what I was trying to get at. That's fine then.